modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AttachmentAwareClient.java

```diff
@@ -62,14 +67,20 @@ public T uploadAndAttachAttachment(T itemToModify, Path fileToAttach, String kin
 
     private T uploadAndAttachAttachment(T itemToModify, Path fileToAttach, HttpEntity<MultiValueMap<String, Object>> requestEntity) {
         final String self = itemToModify.get_Links().getSelf().getHref();
-        ResponseEntity<T> response = restTemplate.postForEntity(self + ATTACHMENTS_ENDPOINT, requestEntity, getHandledClassType());
+        ResponseEntity<T> response;
+        try {
+            response = restTemplate.postForEntity(self + ATTACHMENTS_ENDPOINT, requestEntity, getHandledClassType());
+        } catch (HttpClientErrorException | HttpServerErrorException e) {
+            LOGGER.warn("Request to get attach {} to {} failed with {}", fileToAttach, self, e.getStatusCode());
+            return itemToModify;
+        }
 
         if (response.getStatusCode().is2xxSuccessful()) {
             return Optional.ofNullable(response.getBody())
                     .orElseThrow(() -> new ExecutionException("Body was null"));
```

Comment:
`itemToModify` is the original item. Or do you mean something else?

Reply:
Yes, itemToModify

Reply:
Would it be okay for you, if I include the null check in the first if statement here? since the outcome is pretty much the same?

Reply:
Why not .orElse(itemToModify) instead of the orElseThrow? Cool would be one functional sequence of using only the Optional with a closing orElse? But not necessarily better readable.




Definitely cooler 👿